c7n/policy.py

```diff
@@ -80,7 +80,7 @@ def __init__(self, data, options):
             if 'all' in options.regions:
                 options.regions = all_regions
             for region in options.regions:
-                if region not in all_regions:
+                if len(all_regions) > 0 and region not in all_regions:
```

Comment:
This is a duct tape patch to get c7n working again. It should be removed when a better fix is put in place.